### PR TITLE
Add support for GitHub's Fetch API polyfill

### PR DESCRIPTION
**Pull request by ryan-codingintrigue**
Adding typedef for https://github.com/github/fetch

### CODE REVIEW

**Comment by vvakame** [comment]
@ryan-codingintrigue thank you your contributions.
please check this document. http://definitelytyped.org/guides/contributing.html#naming-the-file
rename fetch/fetch.d.ts to whatwg-fetch/whatwg-fetch.d.ts.

File: fetch/fetch.d.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+﻿// Type definitions for fetch API
+// Project: https://github.com/github/fetch
+// Definitions by: Ryan Graham <https://github.com/ryan-codingintrigue>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/// <reference path="../typescript-services/typescriptServices.d.ts" />
```
Comment (vvakame) [inline fetch/fetch.d.ts]: why need this lines?

Comment (ryan-codingintrigue) [inline fetch/fetch.d.ts]: The `typescriptServices`? Needed as `Headers` is iterable and therefore implements `TypeScript.Iterator<string>`. Unless I'm using that wrong...

Comment (vvakame) [inline fetch/fetch.d.ts]: I think TypeScript is not directly related to the Fetch API...

Comment (ryan-codingintrigue) [inline fetch/fetch.d.ts]: My bad, I thought that was an ES6 iterator implementation, I'll remove that

**Comment by vvakame** [comment]
@ryan-codingintrigue thanks mate :+1:

**Pull request merged**